geoengine/auth.py

```diff
@@ -36,7 +36,7 @@ class Session:
     __server_url: str
     __timeout: int = 60
 
-    session: ClassVar[req.Session] = None
+    session: ClassVar = None
```

Comment:
why is the type removed?

Reply:
I'm still trying to optimize this one, but the gist of it is, that there is no object of type `requests.Session` used anywhere.
All the calls to req inside the `__init__` function are returning json results.
The only time the actual class variable is set is here: 
https://github.com/geo-engine/geoengine-python/blob/b7dce883d39eb0f6edaeecf4df25a632eb09f79b/geoengine/auth.py#L154

But this is also not an object of type `requests.Session`.  
Essentially, removing the type here silences the mypy error for the time being without breaking the code.

Reply:
Maybe the `Session` class should be remodeled as a singleton?

Reply:
Well, the constructor `Session(server_url, credentials, token) ` should return an object of type `Session`, right?

Reply:
True, but not possible with default python. I did not know that this can be fixed with `from __future__ import annotations`. Should be fixed with [6ef3de4](https://github.com/geo-engine/geoengine-python/pull/70/commits/6ef3de48421602fa8eee228d0851b29eebe91b47)